### DISCUSSION TITLE
LunarVim is slower than VSCode

### DISCUSSION THREAD

**Discussion by RizaHKhan** [LunarVim/LunarVim#3359] [General]
I don't even have that many plugins, but searching file, moving around in files. Everything is slow. 

As apposed to VSCode which seems much faster. Has anyone also experienced this and how did you fix it?

**Comment by kobzar** [comment]
I can approve this only for windows OS. I remember that when i try to run LunarVim on Windows it was terrible. It was like chess.
For *nix like OS - all working faster that any code editor but slovly as clear neovim. 
The next one problem with freezes for me was colorizer plugin.

**Comment by codingluke** [comment]
On a Mac M1-Max I can confirm this. But not only LunarVim, Neovim itself seems to be slower, especially on larger files, even with kitty.  The VsCode folks did a huge job here, given its TypeScript fundation. However, I use neovim for its other benefits, the speed is good enough IMHO.

**Comment by danielo515** [comment]
I also noticed certain slowdown on my M1 mac recently, but as @codingluke , I use LunarVim for its other benefits, and speed is just good enough

**Comment by riza-khan** [comment]
Tried with [Alacritty](https://github.com/alacritty/alacritty) (on a Intel chipped Mac) and speed has drastically improved for me. Having said that Alacritty, it turns out, is a pretty niche terminal and some users may find its limitations... unacceptable.

**Reply by dimitarvp** [reply]
Can confirm that Alacritty improves things significantly on an Intel Mac.

**Comment by j0of** [comment]
hey, i'm on windows, and lunarvim performs really badly for me as well. have a found a solution yet?

**Reply by kobzar** [reply]
This is not luanvim or any another neovim pack problem.
You can run clear neovim and catch the same lagged situation.
So, if you want use vim from Windows - use DOcker or WSL

**Reply by vaibhav3027** [reply]
i face this issue in ubuntu as well, all vim, neovim, lunar every terminal editor is really slow, it takes few seconds for every keystrokes to register, sometimes it doesn't register any keystroke until i press another key, so if i have pressed backspace multiple times nothing deletes until i type another word, especially on vim command mode, like if i press :wq to quit vim.
it does nothing until i press another key, i have to press arrow for the previous keystroke to render, without changing the command i have entered


PS: vscode and even android studio works smoothly
Config: i5 8400, 24 GB RAM, Nvidia RTX 2060, OS on SSD

**Comment by Oussa-Err** [comment]
I can confirm this issue on Windows 10. I've observed that neovim becomes noticeably slow due to very high RAM consumption
<img width="775" height="79" alt="image" src="https://github.com/user-attachments/assets/0f6c0f48-19e5-4111-a5cc-683f80360228" />

**Comment by Sloan-Yang** [comment]
❤
